src/test/rpc/Book_test.cpp

```diff
@@ -800,6 +800,460 @@ class Book_test : public beast::unit_test::suite
+        env.trust( USD(1000), "alice" );
+        env.trust( USD(1000), "bob" );
+        env(pay(gw, "alice", USD(100)));
+        env(pay(gw, "bob",   USD(50)));
+        env(offer("alice", drops(4000), USD(10)));
+        env.close();
+
+        Json::Value jvParams;
+        jvParams[jss::taker] = env.master.human();
+        jvParams[jss::taker_pays][jss::currency] = "XRP";
+        jvParams[jss::ledger_index] = "validated";
+        jvParams[jss::taker_gets][jss::currency] = "USD";
+        jvParams[jss::taker_gets][jss::issuer] = gw.human();
+        auto const jrr =
+            env.rpc("json", "book_offers", to_string(jvParams)) [jss::result];
+        env.close();
+        BEAST_EXPECT(jrr[jss::offers].isArray());
+        BEAST_EXPECT(jrr[jss::offers].size() == 1);
+        auto const jrOffer = jrr[jss::offers][0u];
+        BEAST_EXPECT(jrOffer[sfAccount.fieldName] == Account("alice").human());
+        BEAST_EXPECT(jrOffer[sfBookDirectory.fieldName] ==
+            "1C5C34DB7DBE43E1EA72EE080416E88A87C18B2AD29BD8C4570E35FA931A0000");
+        BEAST_EXPECT(jrOffer[sfBookNode.fieldName]       == "0000000000000000");
+        BEAST_EXPECT(jrOffer[jss::Flags]                 == 0);
+        BEAST_EXPECT(jrOffer[sfLedgerEntryType.fieldName] == "Offer");
+        BEAST_EXPECT(jrOffer[sfOwnerNode.fieldName]      == "0000000000000000");
+        BEAST_EXPECT(jrOffer[sfSequence.fieldName]       == 3);
+        BEAST_EXPECT(jrOffer[jss::TakerGets] == USD(10).value().getJson(0));
+        BEAST_EXPECT(jrOffer[jss::TakerPays] == drops(4000).value().getJson(0));
+        BEAST_EXPECT(jrOffer[jss::index] ==
+            "2A432F386EF28151AF60885CE201CC9331FF494A163D40531A9D253C97E81D61");
+        BEAST_EXPECT(jrOffer[jss::owner_funds] == "100");
+        BEAST_EXPECT(jrOffer[jss::quality]     == "400");
+
+        BEAST_EXPECT(wsc->findMsg(5s,
+            [&](auto const& jv)
+            {
+                auto const& t = jv[jss::transaction];
+                return t[jss::TransactionType] == "OfferCreate" &&
+                       t[jss::TakerGets]   == USD(10).value().getJson(0) &&
+                       t[jss::owner_funds] == "100" &&
+                       t[jss::TakerPays]   == drops(4000).value().getJson(0);
+                return false;
+            }));
+
+        env(offer("bob", drops(2000), USD(5)));
+        env.close();
+
+        BEAST_EXPECT(wsc->findMsg(5s,
+            [&](auto const& jv)
+            {
+                auto const& t = jv[jss::transaction];
+                return t[jss::TransactionType] == "OfferCreate" &&
+                       t[jss::TakerGets]   == USD(5).value().getJson(0) &&
+                       t[jss::owner_funds] == "50" &&
+                       t[jss::TakerPays]   == drops(2000).value().getJson(0);
+                return false;
+            }));
+
+        BEAST_EXPECT(wsc->invoke("unsubscribe",
+            books)[jss::status] == "success");
+    }
+
+    void testBookOfferErrors(void)
```

Comment:
Nit: We usually don't specify `(void)` in the parameter list. We usually write `()`